src/mds/MDSMap.h

```diff
@@ -634,6 +658,8 @@ class MDSMap {
 
   mds_rank_t tableserver = 0;   // which MDS has snaptable
   mds_rank_t root = 0;          // which MDS has root directory
+  std::unordered_set<mds_gid_t> qdb_cluster_members;
```

Comment:
When I fix the problem with the messaging sessions, I will want to expand my cluster to standy-replays. To do this consistently, I need the cluster to be computed by the mds monitor. It's just part of my required cluster state, so it's not a duplication.

Also, the leader calculation should be in sync with the members set, it would be awkward to have one and the other calculated at different places.

Reply:
> When I fix the problem with the messaging sessions, I will want to expand my cluster to standy-replays. To do this consistently, I need the cluster to be computed by the mds monitor. It's just part of my required cluster state, so it's not a duplication.

Even if you were to expand the cluster to include standby-replays (which I don't agree with), the MDSMap includes those daemons so the MDS could still manage cluster membership without the help of the monitors.

> Also, the leader calculation should be in sync with the members set, it would be awkward to have one and the other calculated at different places.

The client/mons do not need to know how the cluster membership is maintained. The client will only talk to the leader. We do not need the members in the map.

Reply:
We discussed this at length and decided that it's better to keep the membership in the map to avoid split brain scenarios. There weren't any acceptable alternatives considered.